Hw10

chilik1020 opened the pull request.

chilik1020 committed:
Created layouts

chilik1020 committed:
Added Adapter, ClickListeners

chilik1020 committed:
MusicPlayer interface, NotificationUtil

chilik1020 committed:
Added SongAdapter

chilik1020 committed:
ImageViewCurrentPlaying song

chilik1020 committed:
Changed SongProvider

chilik1020 committed:
Added Notification layuot

chilik1020 committed:
Notification action handle

chilik1020 committed:
Added MusicNotificationManager interface

chilik1020 committed:
Fix ImageView current song and Button play/pause

chilik1020 committed:
Added NotificationPlayerAction enum class

chilik1020 committed:
Added Close button in Notification

chilik1020 committed:
Closing activity onClose notification

DenisAndroidCource commented on hw10/src/main/java/com/chilik1020/hw10/presentation/MainActivity.kt:
```diff
@@ -0,0 +1,170 @@
+        override fun onServiceConnected(name: ComponentName, service: IBinder) {
+            musicService = (service as MusicService.MusicBinder).getMusicService()
+            isBound = true
+            musicPlayer = musicService.musicPlayer
+            musicPlayer.setPlayList(songList)
+            musicPlayer.currentSongLiveData.observe(this@MainActivity) { updateCurrentSong(it) }
+            musicPlayer.playerStatus.observe(this@MainActivity) { changeBtnIconDrawable() }
```
тут можно обойтись скуоп функцией with

DenisAndroidCource commented on hw10/src/main/java/com/chilik1020/hw10/presentation/MainActivity.kt:
```diff
@@ -0,0 +1,170 @@
+        }
+
+        override fun onServiceDisconnected(name: ComponentName?) {
+            isBound = false
```
я бы тут занулил еще и ссылку на сервис

DenisAndroidCource commented on hw10/src/main/java/com/chilik1020/hw10/presentation/MainActivity.kt:
```diff
@@ -0,0 +1,170 @@
+            serviceConnection,
+            Context.BIND_AUTO_CREATE
+        )
+        startService(musicServiceIntent)
```
Context.BIND_AUTO_CREATE означает, что сервис будет запущен, если он не работает в данный момент. Так что startService тут лишний

chilik1020 committed:
MusicService and MusicPlayer as nullable in MainActivity. Added scope functions

chilik1020 committed:
Merge branch 'dev' into hw10
